### PR TITLE
gitkraken: 9.9.2 -> 9.10.0

### PR DESCRIPTION
**Pull request by nicolas-goudry**
## Description of changes

See [official release notes](https://help.gitkraken.com/gitkraken-client/current/#version-9-10-0).

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/version-management/gitkraken/default.nix [hunk 1]
```diff
@@ -10,24 +10,24 @@ with lib;
...
-  version = "9.9.2";
+  version = "9.10.0";
 
   throwSystem = throw "Unsupported system: ${stdenv.hostPlatform.system}";
 
   srcs = {
     x86_64-linux = fetchzip {
       url = "https://release.axocdn.com/linux/GitKraken-v${version}.tar.gz";
-      sha256 = "sha256-UfzHkgqxEaSsoiDwFLsyIBW2min9AvSBrLPJ2MlKh3U=";
+      sha256 = "sha256-JVeJY0VUNyIeR/IQcfoLBN0I1WQNFy7PpCjzk5bPv/Q=";
```
Comment (kirillrdy) [inline pkgs/applications/version-management/gitkraken/default.nix:20]:
```suggestion
      hash = "sha256-JVeJY0VUNyIeR/IQcfoLBN0I1WQNFy7PpCjzk5bPv/Q=";
```

File: pkgs/applications/version-management/gitkraken/default.nix [hunk 2]
```diff
@@ -10,24 +10,24 @@ with lib;
...
     };
 
     x86_64-darwin = fetchzip {
       url = "https://release.axocdn.com/darwin/GitKraken-v${version}.zip";
-      sha256 = "sha256-ble0n+giM8xmuSewBVdj+RuT2093rW0taNzsyQLO92I=";
+      sha256 = "sha256-npc+dwHH0tlVKkAZxmGwpoiHXeDn0VHkivqbwoJsI7M=";
```
Comment (kirillrdy) [inline pkgs/applications/version-management/gitkraken/default.nix:25]:
```suggestion
      hash = "sha256-npc+dwHH0tlVKkAZxmGwpoiHXeDn0VHkivqbwoJsI7M=";
```

File: pkgs/applications/version-management/gitkraken/default.nix [hunk 3]
```diff
@@ -10,24 +10,24 @@ with lib;
...
     };
 
     aarch64-darwin = fetchzip {
       url = "https://release.axocdn.com/darwin-arm64/GitKraken-v${version}.zip";
-      sha256 = "sha256-QYhYzjqbCO0/pRDK7c5jYifj+/UY7SLpRqQUQ3LBFkE=";
+      sha256 = "sha256-fszsGdNKcVgKdv97gBBf+fSODzjKbOBB4MyCvWzm3CA=";
```
Comment (kirillrdy) [inline pkgs/applications/version-management/gitkraken/default.nix:30]:
```suggestion
      hash = "sha256-fszsGdNKcVgKdv97gBBf+fSODzjKbOBB4MyCvWzm3CA=";
```

**Comment by kirillrdy** [comment]
@nicolas-goudry please update commit messages to match contribution guidelines https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md

**Comment by nicolas-goudry** [comment]
> @nicolas-goudry please update commit messages to match contribution guidelines https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md

Sorry about that, it should be ok now.

**Comment by kirillrdy** [comment]
> > @nicolas-goudry please update commit messages to match contribution guidelines https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
> 
> Sorry about that, it should be ok now.

no need to apologises

Going to few a few days to hear back from existing maintainers

**Comment by kirillrdy** [comment]
> @kirillrdy It seems there is an [issue with Gitkraken on NixOS](https://github.com/NixOS/nixpkgs/issues/267434#issuecomment-1842901179), unrelated to this PR, but still.
> 
> Should I put this PR in draft until I can include the relevant fix? Or should @sheepforce create a PR to fix the issue on its own?

not sure if its same issue what what is described https://github.com/NixOS/nixpkgs/issues/267434

but for me, commit graph didnt render because by default gitkranen bundles own git, which doesnt run on nixos, in preferences change to use git from nixos fixes the issue

image

**Comment by nicolas-goudry** [comment]
Indeed I had to change the settings too to use the OS git binary. Forgot to mention that to @sheepforce…

However, it still seems (from this comment) that they have an issue related to `libGL` missing from the package `buildInputs`. Not sure it’s related though. Let me give them a heads up in the original issue to avoid polluting the PR.

In any case, if this needs fixing, it should belong in another PR, right?

**Comment by kirillrdy** [comment]
> Indeed I had to change the settings too to use the OS git binary. Forgot to mention that to @sheepforce…
> 
> However, it still seems (from this comment) that they have an issue related to `libGL` missing from the package `buildInputs`. Not sure it’s related though. Let me give them a heads up in the original issue to avoid polluting the PR.
> 
> In any case, if this needs fixing, it should belong in another PR, right?

up to you, i don't have strong preference.

as for git issue, it should be possible to patch gitkranen so that it always uses git from nixpkgs.

**Comment by nicolas-goudry** [comment]
> it should be possible to patch gitkranen so that it always uses git from nixpkgs

That would be so great. I’m not at all an expert in nix software packaging, but I did try to mess with the `installPhase` to link the `git` binary to the bundled `git` binary of Gitkraken (located under `share/resources/app.asar.unpacked/git/bin/git`) without success, I always get a `Permission denied` error. The files belong to `root`, and it seems that the user is `nixbld` in this phase.

Not sure how to do that. Did you have something particular in mind?

**Comment by nicolas-goudry** [comment]
Ok that did the trick, thank you very much. How is it that the same commands didn’t work in the `installPhase`? I didn’t find anything relevant in nixpkgs documentation.

While this is working, I’m not sure it’s a good idea to patch the git binary to use the one from nixpkgs, since in the UI the version stated is wrong:

Screenshot from 2023-12-07 07-28-44

```
/nix/store/z2h6ndg9sh9v0ni0j0821w6q8wb8hmff-gitkraken-9.10.0/share/gitkraken/resources/app.asar.unpacked/git/bin/git --version  
git version 2.42.0
```

One way that I think we could do that is to force the bundled `git` binary version to be the same as the one stated in the UI. I think it might be something hardcoded in code by Gitkraken developers. What do you think?

**Comment by kirillrdy** [comment]
> Ok that did the trick, thank you very much. How is it that the same commands didn’t work in the `installPhase`? I didn’t find anything relevant in nixpkgs documentation.
> 
> While this is working, I’m not sure it’s a good idea to patch the git binary to use the one from nixpkgs, since in the UI the version stated is wrong:
> 
> Screenshot from 2023-12-07 07-28-44
> 
> ```
> /nix/store/z2h6ndg9sh9v0ni0j0821w6q8wb8hmff-gitkraken-9.10.0/share/gitkraken/resources/app.asar.unpacked/git/bin/git --version  
> git version 2.42.0
> ```
> 
> One way that I think we could do that is to force the bundled `git` binary version to be the same as the one stated in the UI. I think it might be something hardcoded in code by Gitkraken developers. What do you think?

I think version mismatch is better than not working git out of the box.
I think it should also be possible to patch what is displayed in that dropdown

as per installPhase vs postPatch, same permissions apply to both. hence `rm -rf`

**Comment by nicolas-goudry** [comment]
I included the `git` binary patch to the PR. Didn’t fix the UI yet, I’ll try to make it work locally and provide a separate PR if I manage to do it.

I won’t make any further (unrequested) changes to this PR.

**Comment by nicolas-goudry** [comment]
@kirillrdy I don’t understand why the `ofborg-eval` check failed. Everything is working fine when I build. Is it because I made several changes to the PR?

**Comment by kirillrdy** [comment]
@nicolas-goudry don't worry that, eval error happens to all PRs, some attribute got removed without being properly checked

**Comment by kirillrdy** [comment]
@nicolas-goudry it should be fixed , just rebase on latest master

**Review by flexiondotorg — APPROVED** [review]
I confirm this works, and the follow-on pull request should also be merged too: https://github.com/NixOS/nixpkgs/pull/273098

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by kirillrdy** [comment]
@NickCao sorry for assigning reviews to you, I'll limit myself to 2 per week

**Review by NickCao — APPROVED** [review]
diff LGTM, tested basic functionality, apart from the existing issue around libGL looks fine.

**Pull request merged**